packages/iov-bns/src/bnsconnection.ts

```diff
@@ -521,14 +520,14 @@ export class BnsConnection implements BcpAtomicSwapConnection {
    * A helper that triggers if the balance ever changes
    */
   public changeBalance(addr: Address): Stream<number> {
-    return this.changeTx([bnsFromOrToTag(addr)]);
+    return this.changeTx({ address: addr });
   }
 
   /**
    * A helper that triggers if the nonce every changes
    */
   public changeNonce(addr: Address): Stream<number> {
-    return this.changeTx([bnsNonceTag(addr)]);
+    return this.changeTx({ tags: [bnsNonceTag(addr)] });
```

Comment:
what about adding a nonce tag? Or maybe a `signedBy: Address` field to the TxQuery, which this really is trying to do

Reply:
Postponed this for now: https://github.com/iov-one/iov-core/issues/714